### ISSUE TITLE
facebook

### ISSUE THREAD

**Issue by ghost** [boxeehacks/boxeehack#221]
facebook share doesn't work anymore for me this day I reconnected trough the boxee box website and reinstald boxee app on facebook didn't work for me can some chek this out if it works for them thanks

**Comment by ghost** [comment]
where can I donate

**Comment by boxeehacks** [comment]
Checkout the donate button on boxeeplus.com ;)

**Comment by boxeehacks** [comment]
Anyone else have this issue? ... can't test because I don't have it linked up to Facebook...

If not I'll try and test this later, but can take a while...

**Comment by helgeim** [comment]
I have the same problem.

**Comment by walesmd** [comment]
_Confirmed_. The database that keeps track of these services is located at `/data/.boxee/UserData/profiles/walesmd/Database/boxee_window_state.db`. I had one record in the DB (Twitter); finally found where the Share option is when viewing a movie (holy hell that's impossible to find - add it to the list!).

Twitter credentials were still there and "supposedly" valid, added my Facebook, neither service posted. When checking the authorized apps at each service: Boxee did not grant itself permission to do anything to my account (although I authorized it too), there's nothing related to Boxee or D-Link listed. Twitter, I have a Boxee by Boxee on tvv app with some hideous icons but it didn't post my status update either.

Running slim on time right now so I won't be able to track this down any further tonight but it appears as if it will be a bit more than just changing a few URL strings thanks to ever-changing APIs. I wouldn't be surprised if this will require the creation of an all new app on each of these services and top-down refactoring.

To see your stored services:

```
# cd /data/boxee/UserData/profiles/[Your Name]/Database
# sqlite3 -interactive -column -header boxee_window_state.db
# select * from userServices;
# .quit
```

**Comment by ghost** [comment]
Your request (#66045) has been updated. 

To review the status of the request and add additional comments, follow the link below:
http://support.boxee.tv/tickets/66045

You can also add a comment by replying to this email.

---

marcel, Dec 20 06:50 am (EST): 
Thanks will try and get a fix asap

---

Soufian Elamriti, Dec 20 06:41 am (EST): 
ok thanks I have alsow tried to reconnect facebook on boxee website and reinstall boxee app on facebook doesn't work so but thanks for the rep

---

marcel, Dec 19 09:59 am (EST): 
Hi

Thanks for the information we will look into this issue and try and fix it asap

Sorry for the inconvenience

Marcel

---

Soufian Elamriti, Dec 16 06:46 am (EST): 
couple of weeks ago the sharing stopt working for me I don't know why I can share videos and tv shows on my boxee box but when I look at my facebook page it doesn show I shared nothing on it ??? I did a reset today installed every thing firmware etc shared a video with my facebook but it shows nothing ?? why is that bug ???

**Comment by walesmd** [comment]
TL;DR: Facebook is reading fine, writing is broke hard. Technically, very likely we could fix it with minimal effort although it will require a conscious decision and establishment of a "position" from the collective whole as to where we are taking Boxee+. The solution to the Facebook issue, and many others, can only be resolved through firm, public actions that solidifies Boxee+'s position as an independent fork of an ignored product (I'm unsure if this has already been established as there is little in regards to a Boxee+ roadmap or way-ahead).

I did some more investigation on this, trying to track down where the point of failure is after logging into my Facebook Apps control panel on a whim.
1. [Boxee](https://www.facebook.com/apps/application.php?id=176422717603)(links to FB App) is within my authorized users list and had been used within the last 24 hours, it is also authorized to post on my behalf.
2. When visiting the old Boxee Box site it does appear as if it is still polling Facebook for activity within my stream as it's showing videos friends have recently posted. Facebook's Access Log shows the following activity in the last 90 days: Basic Info, Likes/Music/TV/Movies/etc and News Feed were accessed today; About Me/Education/etc was yesterday.
3. The status updates you enter via the Boxee Box are showing up on the old Boxee Box site under "My activity".
4. Looking at my profile I don't see any posts from Boxee in my News Feed, particularly the status update I did yesterday as a test. Scrolling through my Activity Log, including activities only visible to myself, I do not see anything Boxee has posted there. Filtering by the Boxee application itself shows no activity (I never had FB turned on until yesterday to track this bug down).

**Analysis:** It appears as if Boxee's utilization of the Facebook API has simply degraded over time, which is to be expected when you completely ignore a product for a great period of time. Without going through archived Facebook API documentation and comparing it to the Facebook Open Graph API it is still pretty obvious that the _old_ GET calls to the Facebook API that Boxee was using are still returning relevant data as are the calls to associate a Facebook account with the Boxee service which would have also been GET calls. Any API call that results in the creation/modification/deletion (POST/PUT/DELETE) of data on Facebook's system is no longer functioning (posting to walls, posting of activities - which are relatively new when compared to Boxee). This is further confirmed by the fact that the Boxee application is not listed in the left-hand menu within the "main" portion of Facebook - this left hand menu is typically reserved for application that are fully compliant with the Facebook Open Graph API. Deeper within the Facebook preferences we find Boxee listed with all of our other applications, which include very minor Facebook applications such as simply logging in to comment services (Disqus, for instance).

**Recommendation:** The most optimal solution is also the most unlikely, a transfer of the Boxee Facebook application from its current ownership to a Boxee+ maintainer. The most likely solution is still easily obtainable within a relatively short period of time though: a Boxee+ application for Facebook should be registered at which time, depending on our technical issues with the bootloader and where this particular segment of code exists, we'd be able to update the Boxee+ application to once again support the sharing of what you are watching.

**Comment by boxeehacks** [comment]
Boxee+ is not so much an independent fork as a product that builds on top of what is there. This is mainly because of technical reasons though. We cannot compile our own version of the Boxee executable because we're missing parts of the required Intel SDK. As long as we don't have that we cannot modify many of the core parts and can only work around and/or build on top of the existing stuff.

This means that if something is broken with a part of the implementation that is in the Boxee executable itself, then we cannot fix it.

This includes this issue...

Wish there was more we could do...

**Comment by boxeehacks** [comment]
The Facebook part is partly in the Boxee executable, and partly part of the boxee server btw. Neither of which we can fix at the moment. There are plans for a complete server replacement (some people on the forums have been working on that too). If and when that happens maybe we can solve some of these types of issues ourselves. For now however unfortunately not :(
